docs/on_chain_views.md

```diff
@@ -0,0 +1,179 @@
+                           VIEW "fib" nat;
+                           IF_SOME { ADD; } { FAIL }
+                         }
+                         { FAIL };
+               }
+          }
+     }
+```  
+  
+## Calling an on-chain view
+
+Views are meant to be called by a contract using the Michelson Instruction `View` followed by the view name and its result type.
+
+**Example of a contract script having a call to a view named `fib`:**
+
+```
+parameter (pair nat address) ;
+storage nat;
+code {
+       CAR;
+       UNPAIR;
+       VIEW "fib" nat;
+       IF_SOME {NIL operation ; PAIR;} { FAIL }
+     }
+```
+
+**Example of calling a contract entrypoint that makes a call to a view using Taquito:**
+
+The following live code example shows a contract (`contractCallFib`) calling the view `fib` of another contract (`contractTopLevelViews`). 
+
+The example first shows the initial storage of the contract `contractCallFib`. It calls the default entry point of `contractCallFib` with the value of its storage + 1 and the address of the contract `contractTopLevelViews`. A call is made the `fib` view of `contractTopLevelViews` with the `storage + 1` as argument. The view returns the value of the Fibonacci sequence at the position represented by `storage + 1`. The storage of `contractCallFib` is updated to the result of the view.
+
+```js live noInline
+const contractTopLevelViews = 'KT1KJbtwARB2w3yN5fRjBE6CXJh5t9Hbbbpv';
+const contractCallFib = 'KT1T8FPwfbfWK5ir7A5bj6coqttKwhwchdnD';
+
+Tezos.contract.at(contractCallFib)
+  .then((contract) => {
+    contract.storage()
+      .then((storage) => {
+        println(`The initial storage of ${contractCallFib} is :${storage}.`);
+        const fibPosition = storage.toNumber() + 1;
+        println(`Calling the default method of ${contractCallFib} will call the view fib of ${contractTopLevelViews} with ${fibPosition}.`);
+        return contract.methodsObject.default({ 0: fibPosition, 1: contractTopLevelViews }).send()
+          .then((op) => {
+            println(`Waiting for ${op.hash} to be confirmed...`);
+            return op.confirmation().then(() => op.hash)
+              .then(() => {
+                return contract.storage()
+                  .then((finalStorage) => {
+                    println(`The storage is now ${finalStorage} which correspond to the value of the Fibonacci sequence at position ${fibPosition}.`);
+                  })
+              })
+          })
+      })
+  })
+  .catch((error) => println(`Error: ${JSON.stringify(error, null, 2)}`));
+```
+
+## How to simulate a view execution using Taquito
+
+Taquito offers the ability to simulate the result of on-chain views. 
+
+The user can create an instance of `ContractAbstraction` using the `at` method of the Contract or Wallet API with the contract's address that defines the views. The `contractViews` member of the `ContractAbstraction` instance is dynamically populated with methods that match the on-chain view names. 
+
```

Comment:
" view arguments in parameter and "...    

should it be "the parameter",  ?

Reply:
I think my sentence is not really clear :P

It is better:
*`contractViews` is an object where the key is the view name, and the value is a function that takes the view arguments as a parameter and returns an instance of `OnChainView` class.*